src/cephadm/cephadm

```diff
@@ -3571,22 +3589,30 @@ class CephadmAgent():
         self.ack = 1
         self.event = Event()
         self.mgr_listener = MgrListener(self)
-        self.ls_gatherer = AgentGatherer(self, self._get_ls, 'Ls')
-        self.volume_gatherer = AgentGatherer(self, self._ceph_volume, 'Volume')
+        self.ls_gatherer = AgentGatherer(self, lambda: self._get_ls(), 'Ls')
+        self.volume_gatherer = AgentGatherer(self, lambda: self._ceph_volume(enhanced=False), 'Volume')
```

Comment:
don't know if we can do that actually. We've seen really bad issues when running this enhanced scan on unsupported hardware. I fear we have to make this configurable.

Reply:
It still should be configurable with this change. When we first setup the volume gatherer or when the thread listening for messages from the mgr gets new config they call `pull_conf_settings` which updates the volume gatherer func to either use or not use device_enhanced_scan depending on what was found in the config https://github.com/ceph/ceph/pull/44266/files#diff-d0f7acffbce59b9e36a1479d1b1f32955cd811ab8b0b1aa0018a2003f0c4d73eR3692-R3695

Reply:
ok grat! one thing though: we shouldn't run the enhanced scan even a single time, as we've seen super evil hardware bugs.  Is this the case? If yes we can resolve this

Reply:
enhanced defaults to false when we create the volume gatherer object. The only way it's ever set to true (i.e. we call `ceph_volume` with enhanced scan) is if the config we got says it should be which should only happen if the `mgr/cephadm/device_enhanced_scan` config setting is set to true when the cephadm mgr module is creating the config for the agent and since that setting defaults to false there's no way for it to run with enhanced without user intervention